src/components/profile-page/access-tokens/access-token-creation-form.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { ChangeEvent, FormEvent } from 'react'
+import React, { Fragment, useCallback, useMemo, useState } from 'react'
+import { Button, Form } from 'react-bootstrap'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { Trans, useTranslation } from 'react-i18next'
+import { AccessTokenCreatedModal } from './access-token-created-modal'
+import { DateTime } from 'luxon'
+import type { AccessTokenWithSecret } from '../../../api/tokens/types'
+import { postNewAccessToken } from '../../../api/tokens'
+import { showErrorNotification } from '../../../redux/ui-notifications/methods'
+
+interface NewTokenFormValues {
+  label: string
+  expiryDate: string
+}
+
+/**
+ * Form for creating a new access token.
+ */
+export const AccessTokenCreationForm: React.FC = () => {
+  const { t } = useTranslation()
+
+  const expiryDates = useMemo(() => {
+    const today = DateTime.now()
+    return {
+      min: today.toISODate(),
+      max: today
+        .plus({
+          year: 2
+        })
+        .toISODate(),
+      default: today
+        .plus({
+          year: 1
+        })
+        .toISODate()
+    }
+  }, [])
+
+  const tokenDataInitialState = useMemo(() => {
+    return {
+      expiryDate: expiryDates.default,
+      label: ''
+    }
+  }, [expiryDates])
```

Comment:
Just curious.. Why does this memo exist? You need it only in the init function of the following state and it will never change. :thinking:

Reply:
I use it in the `onHideCreatedModal` as well to reset the form value state. The linter suggested using a memo here.